math/geometry.h

```diff
@@ -358,6 +387,184 @@ bool FurthestFreePointCircle(const Eigen::Matrix<T, 2, 1>& line_start,
   return collides;
 }
 
+template <typename T>
+T MinDistanceLineLine(const Eigen::Matrix<T, 2, 1>& a0,
+                      const Eigen::Matrix<T, 2, 1>& a1,
+                      const Eigen::Matrix<T, 2, 1>& b0,
+                      const Eigen::Matrix<T, 2, 1>& b1) {
+  if (CheckLineLineCollision(a0, a1, b0, b1)) {
+    return 0;
+  }
+
+  const auto a0_proj = ProjectPointOntoLineSegment(a0, b0, b1);
+  const auto a1_proj = ProjectPointOntoLineSegment(a1, b0, b1);
+  const auto b0_proj = ProjectPointOntoLineSegment(b0, a0, a1);
+  const auto b1_proj = ProjectPointOntoLineSegment(b1, a0, a1);
+
+  const auto a0_diff_sq = (a0 - a0_proj).squaredNorm();
+  const auto a1_diff_sq = (a1 - a1_proj).squaredNorm();
+  const auto b0_diff_sq = (b0 - b0_proj).squaredNorm();
+  const auto b1_diff_sq = (b1 - b1_proj).squaredNorm();
+
+  const auto min_diff_sq =
+      std::min({a0_diff_sq, a1_diff_sq, b0_diff_sq, b1_diff_sq});
+  return std::sqrt(min_diff_sq);
+}
+
+// Check if line segment collides min and max angle on a circle, moving in a
+// counterclockwise direction from min_angle to max_angle.
+template <typename T>
+T MinDistanceLineArc(const Eigen::Matrix<T, 2, 1>& l0,
```

Comment:
Can you add some unit tests for this? This is a rather complicated function, and we need to ensure that it is correct and that future changes do not break it.

Reply:
I will submit another PR for unit test support and rebase this branch onto that, adding additional tests for this function.